src/Microsoft.DotNet.CodeFormatter.Analyzers/OptimizeNamespaceImportsOptions.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+
+using Microsoft.CodeAnalysis.Options;
+ 
+namespace Microsoft.DotNet.CodeFormatter.Analyzers
+{
+    [Export(typeof(IOptionsProvider)), Shared]
+    internal class OptimizeNamespaceImportsOptions : IOptionsProvider
```

Comment:
Yes, the analyzers can dip into the property bag arbitrarily, or through a private (non-exported) option descriptor. We could also in the future expand PerLanguageOption to have a notion of a public (use-facing) option vs. a hidden (used for test or experimental purposes). Ultimately, the property bag is just a dictionary with some added access/serialization capabilities.

In re: MEF export, yes, understood. Right now, we're building out some ideas here in CodeFormatter. When pushed into Roslyn, we'll need to reexamine them and figure out the appropriate design.